static/js/src/advantage/credentials/dashboard/components/ExamResults/ExamResults.tsx

```diff
@@ -0,0 +1,314 @@
+import {
+  Pagination,
+  Spinner,
+  Notification,
+  ModularTable,
+} from "@canonical/react-components";
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { getExamResults, getIssuedBadgesBulkCredly } from "../../api/queryFns";
+import { ExamResultsMeta, ExamResultsTA, CredlyBadge } from "../../utils/types";
+import BadgeIssueMenu from "./components/BadgeIssueMenu";
+import { getBulkBadgesCredly } from "../../api/queryKeys";
+
+type APIResponse = {
+  results: ExamResultsTA[];
+  meta: ExamResultsMeta;
+  error?: string;
+};
+
+const ExamResults = () => {
+  const queryClient = useQueryClient();
+  const [page, setPage] = useState(1);
+  const [fetchedPages] = useState(new Set([1]));
+  const [cachedData, setCachedData] = useState<Record<number, APIResponse>>({});
+  const [badgeIssuingNotification, setBadgeIssuingNotification] = useState<
+    undefined | "negative" | "positive"
+  >(undefined);
+
+  useEffect(() => {
+    let timeout = null;
+    if (badgeIssuingNotification) {
+      timeout = setTimeout(() => {
+        setBadgeIssuingNotification(undefined);
+      }, 5000);
+    }
+
+    return () => {
+      timeout && clearTimeout(timeout);
+    };
+  }, [badgeIssuingNotification]);
+
+  const onSuccess = (newData: APIResponse) => {
+    if (newData && newData.results) {
+      setCachedData((prev) => ({
+        ...prev,
+        [page]: newData,
+      }));
+    }
+  };
+
+  const { isLoading, isError, data, isFetching } = useQuery({
+    queryKey: ["examResults", page, "finalized"],
+    queryFn: () => getExamResults(page, "finalized", onSuccess),
+  });
+
+  const {
+    isLoading: isLoadingBadges,
+    isError: isErrorBadges,
+    data: dataBadges,
+  } = useQuery<{
+    data: CredlyBadge[];
+  }>({
+    queryKey: getBulkBadgesCredly(),
+    queryFn: () => getIssuedBadgesBulkCredly(),
+    staleTime: Infinity,
+    gcTime: Infinity,
+  });
+
+  useEffect(() => {
+    const queryData = queryClient.getQueryData<APIResponse>([
+      "examResults",
+      page,
+      "finalized",
+    ]);
+
+    if (queryData) {
+      setCachedData((prev) => ({
+        ...prev,
+        [page]: queryData,
+      }));
+    }
+  }, [page]);
+
+  // const handleIssueBadge = async (examId: number) => {};
+
+  const columns: any = useMemo(
+    () => [
+      {
+        Header: "ID",
+        accessor: "id",
+        sortType: "basic",
+        Cell: (props: any) =>
+          props.row.depth === 0 ? (
+            <strong>{props.value}</strong>
+          ) : (
+            <small style={{ paddingLeft: "8px" }}>{props.value}</small>
+          ),
+      },
+      {
+        Header: "User Email",
+        accessor: "user_email",
+        sortType: "basic",
+        Cell: (props: any) =>
+          props.row.depth === 0 ? (
+            <strong>{props.value}</strong>
+          ) : (
+            <small>{props.value}</small>
+          ),
+      },
+      {
+        Header: "Score",
```

Comment:
@usamabinnadeem-10 Can we include the percent in the header so that other Canonical teams know the score is out of 100%? Thanks.

```suggestion
        Header: "Score (%)",
```